### PR TITLE
Feat/linting errors

### PR DESCRIPTION
**Pull request by Ellenn-A**
# Changes:
- removed lint warnings when you run project 

### Ticket: 
- https://digicatapult.atlassian.net/browse/L3-245

### CODE REVIEW

File: src/lib/ipfs.ts [hunk 1]
```diff
@@ -9,6 +9,27 @@ interface FilestoreResponse {
   Hash: string
   Size: string
 }
+type EmptyObject = object
+type Peers = EmptyObject[]
+type PeersResponse = {
+  Peers?: Peers
+}
```
Comment (jonmattgray) [inline src/lib/ipfs.ts:16]: saying Peers is an array of empty objects here but later there's `Peers: { Peer?: unknown }[]` , is it something like 
```
type PeersResponse = {
  Peers?: { Peer: object }[]
}
```

Comment (jonmattgray) [reply src/lib/ipfs.ts:16]: I'm guessing, haven't looked at a real `peersResponse`

File: src/controllers/v1/_common/demand.ts [hunk 2]
```diff
@@ -49,7 +53,10 @@ export class DemandController extends Controller {
       throw new BadRequest('Attachment not found')
     }
 
-    const { address: selfAddress, alias: selfAlias } = await this.identity.getMemberBySelf()
+    const res = await this.identity.getMemberBySelf()
```
[the remark this answers is not in the document]
Comment (Ellenn-A) [reply src/controllers/v1/_common/demand.ts:56]: so doing this brings lint errors back when I run it 
`[0] src/controllers/v1/_common/demand.ts(53,25): error TS18046: 'res' is of type 'unknown'.
[0] 
[0] src/controllers/v1/_common/demand.ts(54,23): error TS18046: 'res' is of type 'unknown'.
[0] 
[0] src/controllers/v1/_common/demand.ts(149,25): error TS18046: 'res' is of type 'unknown'.
[0] 
[0] src/controllers/v1/_common/demand.ts(202,22): error TS18046: 'res' is of type 'unknown'.
[0] 
[0] src/controllers/v1/_common/demand.ts(224,23): error TS18046: 'res' is of type 'unknown'.`

[the remark this answers is not in the document]
Comment (Ellenn-A) [reply src/controllers/v1/_common/demand.ts:56]: and also errors when I change signature on getMemberBySelf 
`  getMemberBySelf = async (): Promise<MemberResponse> => {
    const res = await fetch(`${URL_PREFIX}/v1/self`)

    if (res.ok) {
      return await res.json()
    }
    
   `
   
   Error: `[0] src/lib/services/identity.ts(78,7): error TS2322: Type 'unknown' is not assignable to type 'MemberResponse'.`

File: src/lib/ipfs.ts (hunk 1 above)
Comment (Ellenn-A) [reply src/lib/ipfs.ts]: it passed tests

File: src/controllers/v1/_common/demand.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonmattgray) [reply src/controllers/v1/_common/demand.ts:56]:
```
    if (res.ok) {
      return (await res.json()) as MemberResponse
    }
 ```
 
 should sort that out, reassuring TypeScript that the response is `MemberResponse` and not `unknown`

[the remark this answers is not in the document]
Comment (Ellenn-A) [reply src/controllers/v1/_common/demand.ts:56]: yeah

[the remark this answers is not in the document]
Comment (mattdean-digicatapult) [reply src/controllers/v1/_common/demand.ts:52]: we should generally avoid using `as` in Typescript where possible. We're basically bypassing the type checker which is our friend. A better way is to parse the response with something like `zod`

**Comment by mattdean-digicatapult** [comment]
I've replaced the `ipfs` and `identity` implementations with ones adapted from `sqnc-hyproof-api` which use `zod` for parsing. Note we should abstract these out to shared modules

**Review by mattdean-digicatapult — DISMISSED** [review]

**Review by mattdean-digicatapult — APPROVED** [review]

**Review by dblane-digicatapult — APPROVED** [review]

**Pull request merged**